src/components/Contacts/ContactDetails/ContactDontationsTab/PartnershipInfo/EditPartnershipInfoModal/EditPartnershipInfoModal.tsx

```diff
@@ -0,0 +1,431 @@
+}));
+
+const CheckboxLabel = styled(FormControlLabel)({
+  margin: 'none',
+});
+
+interface EditPartnershipInfoModalProps {
+  contact: ContactDonorAccountsFragment;
+  handleClose: () => void;
+  isOpen: boolean;
+}
+
+export const EditPartnershipInfoModal: React.FC<EditPartnershipInfoModalProps> = ({
+  contact,
+  handleClose,
+  isOpen,
+}) => {
+  const { t } = useTranslation();
+  const accountListId = useAccountListId();
+  const constants = useApiConstants();
+
+  const { enqueueSnackbar } = useSnackbar();
+  const [
+    updateContactPartnership,
+    { loading: updating },
+  ] = useUpdateContactPartnershipMutation();
+  const pledgeCurrencies = constants?.pledgeCurrencies;
+  const method = contact.lastDonation
+    ? contact.lastDonation.paymentMethod
+    : t('None');
+
+  const lastGift = contact.lastDonation
+    ? `${currencyFormat(
+        contact.lastDonation.amount.amount,
+        contact.lastDonation.amount.currency,
+      )}`
+    : '';
+
+  const contactPartnershipSchema: yup.SchemaOf<
+    Pick<
+      ContactUpdateInput,
+      | 'id'
+      | 'status'
+      | 'pledgeAmount'
+      | 'pledgeFrequency'
+      | 'pledgeCurrency'
+      | 'pledgeReceived'
+      | 'pledgeStartDate'
+      | 'nextAsk'
+      | 'noAppeals'
+    >
+  > = yup.object({
+    id: yup.string().required(),
+    status: yup.mixed<StatusEnum>().oneOf(Object.values(StatusEnum)).nullable(),
+    pledgeAmount: yup.number().moreThan(-1).nullable(),
+    pledgeStartDate: yup.string().nullable(),
+    pledgeReceived: yup.boolean().default(false).nullable(),
+    pledgeCurrency: yup.string().nullable(),
+    nextAsk: yup.string().nullable(),
+    noAppeals: yup.boolean().default(false).nullable(),
+    pledgeFrequency: yup.mixed<PledgeFrequencyEnum>().nullable(),
+  });
+
+  const onSubmit = async (
+    attributes: Pick<
+      ContactUpdateInput,
+      | 'id'
+      | 'status'
+      | 'pledgeAmount'
+      | 'pledgeFrequency'
+      | 'pledgeCurrency'
+      | 'pledgeReceived'
+      | 'pledgeStartDate'
+      | 'nextAsk'
+      | 'noAppeals'
+    >,
+  ) => {
+    console.log(attributes.nextAsk);
+    await updateContactPartnership({
+      variables: {
+        accountListId: accountListId ?? '',
+        attributes,
+      },
+    });
+
+    enqueueSnackbar(t('Partnership information updated successfully.'), {
+      variant: 'success',
+    });
+    handleClose();
+  };
+
+  const updateStatus = (
+    status: StatusEnum,
+    setFieldValue: (name: string, value: StatusEnum | number | null) => void,
+  ) => {
+    if (status !== StatusEnum.PartnerFinancial) {
+      setFieldValue('status', status);
+      setFieldValue('pledgeAmount', 0);
+      setFieldValue('pledgeFrequency', null);
+    } else {
+      setFieldValue('status', status);
+    }
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      title={t('Edit Partnership')}
+      handleClose={handleClose}
+    >
+      <Formik
+        initialValues={{
+          id: contact.id,
```

Comment:
In this case, we only want these specific values from the contact since these are the only values that can be edited in this modal. If we added ```...contact```, 1. I think it would throw a type error because the other values are excluded in our validation schema, 2. That would imply to the Formik component that those fields can be edited and will be needed for this form.

Reply:
Ah ok 👍